software/CONFIGURATION.md

```diff
@@ -1,55 +1,136 @@
+- `EUROPI_MODEL` specifies the type of EuroPi module. Currently only `"europi"` is supported
+- `PICO_MODEL` must be one of `"pico"` or `"pico w"`
+- `CPU_FREQ` must be one of `250000000` or `125000000`
+- `ROTATE_DISPLAY` must be one of `false` or `true`
+- `DISPLAY_WIDTH` is the width of the screen in pixels. The standard EuroPi screen is 128 pixels wide
+- `DISPLAY_HEIGHT` is the height of the screen in pixels. The standard EuroPi screen is 32 pixels tall
+- `DISPLAY_SDA` is the I²C SDA pin used for the display. Only SDA capable pins can be selected
+- `DISPLAY_SCL` is the I²C SCL pin used for the display. Only SCL capable pins can be selected
+- `DISPLAY_CHANNEL` is the I²C channel used for the display, either 0 or 1.
+- `MAX_OUTPUT_VOLTAGE` is an integer in the range `[0, 10]` indicating the maximum voltage CV output can generate.
   The hardware is capable of 10V maximum
-- `max_input_voltage` is an integer in the range `[0, 12]` indicating the maximum allowed voltage into the `ain` jack.
+- `MAX_INPUT_VOLTAGE` is an integer in the range `[0, 12]` indicating the maximum allowed voltage into the `ain` jack.
   The hardware is capable of 12V maximum
-- `gate_voltage` is an integer in the range `[0, 12]` indicating the voltage that an output will produce when `cvx.on()` is called
+- `GATE_VOLTAGE` is an integer in the range `[0, 10]` indicating the voltage that an output will produce when `cvx.on()`
+  is called. This value must not be higher than `MAX_OUTPUT_VOLTAGE`
 
 
 
 # Experimental configuration
 
-Other configuration properties are used by [experimental features](software/firmware/experimental/__init__.py)
-and can be set using a similar static configuration file. This file is located at `/config/config_ExperimentalConfig.json` 
+Other configuration properties are used by [experimental features](/software/firmware/experimental/__init__.py)
+and can be set using a similar static configuration file. This file is located at `/config/ExperimentalConfig.json`
 on the Raspberry Pi Pico. If this file does not exist, default settings will be loaded.  The following
 shows the default configuration:
 
 ```json
 {
-    "volts_per_octave": 1.0,
+    "VOLTS_PER_OCTAVE": 1.0,
 }
 ```
 
-- `volts_per_octave` must be one of `1.0` (Eurorack standard) or `1.2` (Buchla standard)
+- `VOLTS_PER_OCTAVE` must be one of `1.0` (Eurorack standard) or `1.2` (Buchla standard)
+
+
+# Accessing config members in Python code
+
+The firmware converts the JSON file into a `ConfigSettings` object, where the JSON keys are converted
+to Python attributes.  The JSON object's keys must follow these rules, otherwise a `ValueError` will be raised:
+
+1. The string may not be empty
+1. The string may only contain letters, numbers, and the underscore (`_`) character
+1. The string may not begin with a number
+1. The string should be in `ALL_CAPS` -- this is not enforced, but is highly recommended
```

Comment:
I wouldn't bother adding the lack of enforcement text, I'm guessing when it comes to it we will actually enforce it, but either way only more advanced programmers/contributors will be adding config values. For users on their own fork they can use whatever they want, but encouraging all caps for the sake of keeping it the same as the official europi documentation is probably for the best

Reply:
Removed.